v3/integrations/nrmongo-v2/nrmongo_test.go

```diff
@@ -0,0 +1,261 @@
+	}
+	se = &event.CommandSucceededEvent{
+		CommandFinishedEvent: finishedEvent,
+		Reply:                nil,
+	}
+	fe = &event.CommandFailedEvent{
+		CommandFinishedEvent: finishedEvent,
+		Failure:              errors.New("failureCause"),
+	}
+	thisHost, _ = sysinfo.Hostname()
+)
+
+func TestOrigMonitorsAreCalled(t *testing.T) {
+	var started, succeeded, failed bool
+	origMonitor := &event.CommandMonitor{
+		Started:   func(ctx context.Context, e *event.CommandStartedEvent) { started = true },
+		Succeeded: func(ctx context.Context, e *event.CommandSucceededEvent) { succeeded = true },
+		Failed:    func(ctx context.Context, e *event.CommandFailedEvent) { failed = true },
+	}
+	ctx := context.Background()
+	nrMonitor := NewCommandMonitor(origMonitor)
+
+	nrMonitor.Started(ctx, ste)
+	if !started {
+		t.Error("started not called")
+	}
+	nrMonitor.Succeeded(ctx, se)
+	if !succeeded {
+		t.Error("succeeded not called")
+	}
+	nrMonitor.Failed(ctx, fe)
+	if !failed {
+		t.Error("failed not called")
+	}
+}
+
+func TestClientOptsWithNullFunctions(t *testing.T) {
+	origMonitor := &event.CommandMonitor{} // the monitor isn't nil, but its functions are.
+	ctx := context.Background()
+	nrMonitor := NewCommandMonitor(origMonitor)
+
+	// Verifying no nil pointer dereferences
+	nrMonitor.Started(ctx, ste)
+	nrMonitor.Succeeded(ctx, se)
+	nrMonitor.Failed(ctx, fe)
+}
+
+func TestHostAndPort(t *testing.T) {
+	type hostAndPort struct {
+		host string
+		port string
+	}
+	testCases := map[string]hostAndPort{
+		"localhost:8080[-1]":                     {host: "localhost", port: "8080"},
+		"something.com:987[-789]":                {host: "something.com", port: "987"},
+		"thisformatiswrong":                      {host: "", port: ""},
+		"somethingunix.sock[-876]":               {host: "somethingunix.sock", port: ""},
+		"/var/dir/path/somethingunix.sock[-876]": {host: "/var/dir/path/somethingunix.sock", port: ""},
+	}
+	for test, expected := range testCases {
+		h, p := calcHostAndPort(test)
+		if expected.host != h {
+			t.Errorf("unexpected host - expected %s, got %s", expected.host, h)
+		}
+		if expected.port != p {
+			t.Errorf("unexpected port - expected %s, got %s", expected.port, p)
+		}
+	}
+}
+
+func TestMonitor(t *testing.T) {
+	var started, succeeded, failed bool
+	origMonitor := &event.CommandMonitor{
+		Started:   func(ctx context.Context, e *event.CommandStartedEvent) { started = true },
+		Succeeded: func(ctx context.Context, e *event.CommandSucceededEvent) { succeeded = true },
+		Failed:    func(ctx context.Context, e *event.CommandFailedEvent) { failed = true },
+	}
+	nrMonitor := mongoMonitor{
+		segmentMap:  make(map[int64]*newrelic.DatastoreSegment),
+		origCommMon: origMonitor,
+	}
+	app := createTestApp()
+	txn := app.StartTransaction("txnName")
+	ctx := newrelic.NewContext(context.Background(), txn)
+	nrMonitor.started(ctx, ste)
+	if !started {
+		t.Error("Original monitor not started")
+	}
+	if len(nrMonitor.segmentMap) != 1 {
```

Comment:
I would suggest checking not just the length but the content of the `segmentMap` here, to make sure it isn't satisfying the length with unexpected/junk data.